Vod Review/video_proccessing.py

```diff
@@ -104,13 +110,82 @@ def kill_marker(frame, debug, meta):
 
     # thresholds frame based on red range
     roi = cv2.inRange(red, low_red, upp_red)
+    if debug is True:
+        cv2.imshow('kill_mark', roi)
 
     # returns an average value of the pixels that are lit
     brightness = np.mean(roi)
 
     # if a certain amount of pixels are lit up it should indicate the kill markers are on screen
     if 4.38 < brightness < 5.1:
         kill = 1
+        if debug is True:
+            print(f'kill:{kill}')
+    else:
+        kill = 0
+
+    return kill
+    
+    
+def kill_marker_test(frame, debug, meta):
+    """ identifies if kill marker present in the frame"""
+
+    # cuts down frame to cross hairs
+    roi = frame[int(int(meta[1]) * 0.4537):
+                int(int(meta[1]) * 0.5462), int(int(meta[0]) * 0.4739): int(int(meta[0]) * 0.526)]
+    roi = cv2.resize(roi, (100, 100))
+    
+    # converts frame to HSV format
+    roi = cv2.cvtColor(roi, cv2.COLOR_BGR2HSV)
+    
+    # range of red to detect in HSV format
+    low_red = np.array([4, 190, 140])
+    upp_red = np.array([7, 230, 250])
+    
+    # blurs it a little to get more consistent colors
+    roi = cv2.GaussianBlur(roi, (3, 3), 0)
+    
+    # thresholds frame based on red range
+    red = cv2.inRange(roi, low_red, upp_red)
+        
+    # finds all contours in red channel
+    mark_cnts = cv2.findContours(red, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
+    mark_cnts = imutils.grab_contours(mark_cnts)
+    
+    # if kill marker is present and clear, only 4 contours should be on screen
+    if len(mark_cnts) != 4:
+        kill = 0
+        return kill
+    
+    # creates mask in the shape of the kill cross hairs
+    mask = np.zeros(roi.shape, np.uint8)
+    
+    # draw contour areas onto the mask image in white, should i draw lines or contours to filter out reds contours potentially not in kill marker locations?

```

Comment:
You can use a linter to automatically pick up on these long lines. PyLint is pretty good from memory, and you can set it up to run as a git pre-commit hook. There is almost certainly a PEP 8 pylint file available for download, which will ensure you meet the style guide that python themselves recommend.